test/integration.test.js

```diff
@@ -52,3 +52,74 @@ test('missing header route fails correctly', (t) => {
     t.error(err)
   })
 })
+
+test('integration with fastify-auth', async (t) => {
+  const fastify = require('fastify')()
+  await fastify.register(plugin, { keys: new Set(['123456']) })
+  await fastify.decorate('allowAnonymous', function (request, _, done) {
+    if (!request.headers.authorization) {
+      return done()
+    }
+    return done(new Error('not anonymous'))
+  })
+  await fastify.register(require('fastify-auth'))
+
+  fastify.route({
+    method: 'GET',
+    url: '/anonymous',
+    preHandler: fastify.auth([
+      fastify.allowAnonymous,
+      fastify.verifyBearerAuth
+    ]),
+    handler: function (_, reply) {
+      reply.send({ hello: 'world' })
+    }
+  })
+
+  await fastify.ready()
+
+  test('anonymous should pass', async (t) => {
```

Comment:
If this is supposed to be a sub-test, then:
```suggestion
  t.test('anonymous should pass', async (t) => {
```